### DISCUSSION TITLE
ERR_SSL_VERSION_OR_CIPHER_MISMATCH 如何处理

### DISCUSSION THREAD

**Discussion by tianshangwuyun** [SpaceTimee/Sheas-Cealer#32] [General]
此网站无法提供安全连接
chatgpt.com 使用了不受支持的协议。
ERR_SSL_VERSION_OR_CIPHER_MISMATCH
协议不受支持
客户端和服务器不支持一般 SSL 协议版本或加密套件。

这类问题如何处理了

**Comment by SpaceTimee** [comment]
这是因为服务器不接受伪造后的 SNI (SNI server_name)，导致 SSL 握手失败

通常这是网站套了 CDN 导致的 (如 Cloudflare)，默认情况下 CDN 会按特定流程严格校验 SNI，证书域名以及 Host 头是否匹配，有这几种解决办法:

1. 可以尝试填一个服务器接受且未被阻断的 SNI，填在伪造规则的第二项参数中 (参考[伪造规则文档的 2](https://github.com/SpaceTimee/Cealing-Host/wiki/Cealing-Host-Documentation))，如果套 Cloudflare 的话，通常填同 DNS Zone 下的其它域名，此外也可以尝试填证书 SAN

2. 修改伪造规则的 IP (第三项参数)，用不严格校验 SNI 且未被封锁的源站 IP 或中转 IP

3. 通知网站管理者修改 CDN 相关设置 (如给 Cloudflare 添加 Cloudflare for SaaS 
自定义主机名) 或去掉 CDN (下下策，通常不易做到)

4. 如果网站不允许不匹配的 SNI server_name 但允许完全没有 SNI (常见于一些套 Cloudfront 的网站，如 Twitch, Niconico 等)，可以尝试 SNI 拔除 (参考 [Special Cealing Instruction](https://github.com/SpaceTimee/Cealing-Host/wiki/Special-Cealing-Instruction))

5. 如果以上方案都不行，通常就无法单靠 SNI 伪造解决了，可以尝试 TLS 切片 (包括 TLS Record 分片，如果网站支持 HTTP/3 的话也可以使用 Client Initial 分片)

你提到的 ChatGPT 是相当难办的，目前应该只能使用 TLS Record 分片直连 (Sheas Cealer 做不到)，另外我不太确定现在 ChatGPT 是否有主动屏蔽大陆 IP，如果主动屏蔽的话可能还需要在此基础上额外使用其他技术

**Reply by sunshineteenager** [reply]
chatgpt是主动拒绝大陆ip的，我在被墙之前访问过，现在也可以用tls分片访问，参考
这个工程：https://github.com/maoist2009/TlsFragment
这是访问截图：
![Screenshot_20250807_045149](https://github.com/user-attachments/assets/26377c9b-43be-4a5d-a756-7346f5bba4d0)
